hledger/Hledger/Cli/Commands/Diff.md

```diff
@@ -1,8 +1,20 @@
 diff\
 Compares two journal files.  It looks at the transactions of a single
 account and prints out the transactions which are in one journal file but not
-in the other.  This is particularly useful for reconciling existing journals
-with bank statements.
+in the other.
+
+This is particularly useful for reconciling existing journals with bank
+statements.  Many banks provide an export of the bank statements, which can be
+converted to ledger files using custom scripts or read directly as CSV files.
+With the diff command you can make sure that these transactions from the bank
+statement match up exactly with the transactions in your ledger file, and that
+the resulting balance is correct.
+
+This command compares the postings of a single account (which needs to have the
+same name in both files), and only checks the amount of the postings (not the
+name or the date of the transactions).  Postings are compared (instead of
+transactions) so that you can combine multiple transactions from the bank
+statement in a single transaction in the ledger file.
```

Comment:
Useful extra detail, thanks. We might tweak the wording some day to make it more generally applicable - I've not yet personally encountered a bank that exports separate "statements", instead I tend to see CSV between two arbitrary dates within the last N months.